Common/Motor.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+namespace sjsu
+{
+/// Abstract interface for devices that drive basic rotational motors.
+/// including motors that that rotate continually such as brushed and brushless DC/AC
+/// motors and excluding stepper and servo motors.
+
+class Motor
+{
+  public:
+  /// Initialize peripherals required to communicate with motor controller. Must be the
+  /// first method called on this driver. After this is called, `Enable()` can
+  /// be called.
+  virtual bool Initilize() = 0;
+  /// This method must be called before running `Speed()`. This function will
+  /// configure the motor controller settings as defined through the constructor of this
+  /// interface's implementation. Some implementations have more detail or
+  /// settings than others.
+  virtual bool Enable() = 0;
```

Comment:
```suggestion
  virtual Returns<void> Enable() = 0;
```
Use the `Returns<void>` return type so that error can be propagated up from driver implementations.